src/services/near-nep21-util.js

```diff
@@ -320,44 +320,44 @@ export async function calcPriceFromOut(tokenProvide, tokenWant) {
   }
 }
 
-export async function swapFromOut(tokenIN, tokenOUT) {
+export async function swapFromOut(tokenProvide, tokenWant) {
 
-  const amountOUT = toYoctosString(tokenOUT.amount); //user input: I want 2 tokens out from the pool -> convert "2" to yoctos
+  const amountWant = toYoctosString(tokenWant.amount); //user input: I want 2 tokens out from the pool -> convert "2" to yoctos
 
-  const amountIN = tokenIN.amount; //computed amount he has to send into the pool, already in yoctos 
+  const amountProvide = tokenProvide.amount; //computed amount he has to send into the pool, already in yoctos 
 
-  if (tokenIN.type === "Native token") {
+  if (tokenProvide.type === "Native token") {
     // Native to NEP-21
```

Comment:
we should add a check here, that `tokenWant.type === "NEP-21"` (and forbid native-native exchange).

Reply:
Also, let's put the string values in a `const`.

Reply:
OK, adding same for NEP-21

Reply:
No, the second case was OK. We support NEP-21 <-> NEP-21 swaps.

Reply:
we can only check if in case of NEP21 swaps, the tokens are different.

Reply:
Yes, i have updated only if:  `token addresses are same then the user can't do swap`